WordPressKitTests/CommentServiceRemoteXMLRPCTests.swift

```diff
@@ -0,0 +1,62 @@
+import XCTest
+import wpxmlrpc
+
+@testable import WordPressKit
+
+class CommentServiceRemoteXMLRPCTests: RemoteTestCase, XMLRPCTestable {
+    private var remote: Any?
+    private let getSiteCommentsSuccessMockFilename = "xmlrpc-site-comments-success.xml"
+
+    override func setUp() {
+        super.setUp()
+        remote = CommentServiceRemoteXMLRPC(api: getXmlRpcApi(),
+                                            username: XMLRPCTestableConstants.xmlRpcUserName,
+                                            password: XMLRPCTestableConstants.xmlRpcPassword)
+    }
+
+    override func tearDown() {
+        super.tearDown()
+        remote = nil
+    }
+
+    // MARK: - Tests
+
+    func testGetCommentsSucceeds() {
+        let expect = expectation(description: "Fetching site comments should succeed")
+
+        stubRemoteResponse(XMLRPCTestableConstants.xmlRpcUrl,
+                           filename: getSiteCommentsSuccessMockFilename,
+                           contentType: .XML)
+
+        if let remoteInstance = remote as? CommentServiceRemote {
+            remoteInstance.getCommentsWithMaximumCount(1,
+                                                       success: { comments in
+
+                guard let comment = comments?.first as? RemoteComment else {
+                    XCTFail("Failed to retrieve mock site comment")
+                    return
+                }
+
+                XCTAssertEqual(comment.author, "Comment Author")
+                XCTAssertEqual(comment.authorEmail, "author@email.com")
+                XCTAssertEqual(comment.authorUrl, "author URL")
+                XCTAssertEqual(comment.authorIP, "000.0.00.000")
+                XCTAssertEqual(comment.link, "comment URL")
+                XCTAssertEqual(comment.parentID, NSNumber(value: 1))
+                XCTAssertEqual(comment.postID, NSNumber(value: 1))
```

Comment:
What do you think about differentiating the mocked values of `parentID` and `postID`? I was thinking this could handle potential cases of wrong assignments.

Reply:
Oh yea that's a good idea. Done!